python/ray/_private/services.py

```diff
@@ -622,7 +628,14 @@ def node_ip_address_from_perspective(address: str):
     Returns:
         The IP address by which the local node can be reached from the address.
     """
-    ip_address, port = address.split(":")
+    ip_address, port = address.rsplit(":", 1)
+    if ip_address == "8.8.8.8":
```

Comment:
Why hard coded? Also, this breaks in IPv6-only environments

Reply:
ok i get it this is for DNS assuming 8.8.8.8 is always available. Is there a better way to resolve DNS?

Reply:
Please let's also add unit tests for this method to ensure it covers both IPv4 and IPv6 IPs.